toffy/qc_comp_test.py

```diff
@@ -178,129 +181,170 @@ def test_compute_99_9_intensity():
-        total_intensity = qc_data['total_intensity']
-        intensity_99_9 = qc_data['99_9_intensity']
+            # write the FOV names in
+            fov_nm_data['fov'] = fov
+            fov_ti_data['fov'] = fov
+            fov_99_9_data['fov'] = fov
 
-        # assert fovs are correct (if fovs is None, set to all fovs)
-        if test_fovs is None:
-            test_fovs = fovs
+            # write the channel names in
+            fov_nm_data['channel'] = chans
+            fov_ti_data['channel'] = chans
+            fov_99_9_data['channel'] = chans
 
-        misc_utils.verify_same_elements(
-            provided_fovs=test_fovs,
-            nzm_fovs=nonzero_mean['fov'].values
-        )
-        misc_utils.verify_same_elements(
-            provided_fovs=test_fovs,
-            ti_fovs=total_intensity['fov'].values
-        )
-        misc_utils.verify_same_elements(
-            provided_fovs=test_fovs,
-            i99_9_fovs=intensity_99_9['fov'].values
-        )
+            fov_nm_data.to_csv(
+                os.path.join(bin_file_path, '%s_nonzero_mean_stats.csv') % fov, index=False
+            )
+            fov_ti_data.to_csv(
+                os.path.join(bin_file_path, '%s_total_intensity_stats.csv') % fov, index=False
+            )
+            fov_99_9_data.to_csv(
+                os.path.join(bin_file_path, '%s_percentile_99_9_stats.csv') % fov, index=False
+            )
 
-        # assert channels are correct (if chans is None, set to all chans)
-        if test_chans is None:
-            test_chans = chans
+        # run the aggregation function
+        qc_comp.combine_qc_metrics(bin_file_path)
 
-        misc_utils.verify_same_elements(
-            provided_chans=test_chans,
-            nzm_chans=nonzero_mean.drop(columns='fov').columns.values
-        )
-        misc_utils.verify_same_elements(
-            provided_chans=test_chans,
-            ti_chans=total_intensity.drop(columns='fov').columns.values
-        )
-        misc_utils.verify_same_elements(
-            provided_chans=test_chans,
-            i99_9_chans=intensity_99_9.drop(columns='fov').columns.values
-        )
+        # assert the correct file names were written
+        combined_nm_path = os.path.join(bin_file_path, 'combined_nonzero_mean_stats.csv')
+        combined_ti_path = os.path.join(bin_file_path, 'combined_total_intensity_stats.csv')
+        combined_p99_9_path = os.path.join(bin_file_path, 'combined_percentile_99_9_stats.csv')
+
+        assert os.path.exists(combined_nm_path)
+        assert os.path.exists(combined_ti_path)
+        assert os.path.exists(combined_p99_9_path)
+
+        # read in the QC data
+        nm = pd.read_csv(combined_nm_path)
+        ti = pd.read_csv(combined_ti_path)
+        p99_9 = pd.read_csv(combined_p99_9_path)
+
+        # assert the column names are correct
+        assert list(nm.columns.values) == ['fov', 'channel', 'Non-zero mean intensity']
```

Comment:
If you create the fake qc data programmatically, rather than randomly, we can assert that it the values are correct as well.

Reply:
I've done this for `combine_qc_metrics`. However it will be difficult to do with `compute_qc_metrics` as none of the test `.bin` files especially with the Gaussian blurring tests.